core/src/main/java/io/questdb/griffin/engine/functions/regex/AbstractLikeVarcharFunctionFactory.java

```diff
@@ -144,13 +148,89 @@ public Function newInstance(
 
     protected abstract boolean isCaseInsensitive();
 
+    /**
+     * Optimized variant of {@link ConstContainsVarcharFunction} with SWAR-based contains implementation.
+     * Works only for patterns up to 8 bytes in size.
+     */
+    private static class ConstContainsSwarVarcharFunction extends BooleanFunction implements UnaryFunction {
+        private static final int MAX_SIZE = Long.BYTES;
+        private final Utf8Sequence pattern; // only used in toPlan
+        private final long patternMask;
+        private final int patternSize;
+        private final long patternWord;
+        private final Function value;
+
+        public ConstContainsSwarVarcharFunction(Function value, Utf8Sequence pattern) {
+            assert pattern.size() > 0 && pattern.size() <= MAX_SIZE;
+            this.value = value;
+            this.patternSize = pattern.size();
+            this.patternMask = patternSize == MAX_SIZE ? -1L : (1L << 8 * pattern.size()) - 1L;
+            long patternWord = 0;
+            for (int i = 0, n = pattern.size(); i < n; i++) {
+                patternWord |= (long) (pattern.byteAt(i) & 0xff) << (8 * i);
+            }
+            this.patternWord = patternWord;
+            this.pattern = pattern;
+        }
+
+        @Override
+        public Function getArg() {
+            return value;
+        }
+
+        @Override
+        public boolean getBool(Record rec) {
+            Utf8Sequence us = value.getVarcharA(rec);
+            if (us == null || us.size() < patternSize) {
+                return false;
+            }
+
+            int size = us.size();
+
+            int i = 0;
+            for (int n = size - MAX_SIZE + 1; i < n; i++) {
```

Comment:
Something to try here is loading a `long` just once, then each loop step shifts it by 8 and loads another byte. This could turn out to have an edge in performance. It would also eliminate the tail as a special case.

Reply:
`checkZeroByte()` will return a `long` that consists of only two kinds of bytes: `00` and `80`, the latter occurring exactly where a zero byte was found. We can combine this with `Long.numbefOfTrailingZeros`, some shifting and AND masks in a loop that checks only the offsets where the byte was found. 

Such code won't contribute much to general performance, but I think it could pay off against worst-case patterns where we keep checking all the bytes (where every 8-byte segment has a matching byte).

Reply:
Tried applying the suggestion in 61569c765918634845372d9703d70ce4f4df54e9. Could you take a look?

Reply:
Cool! Do you think it would also be possible to replace the counted loop with one that iterates over the locations of the found byte in the `long`? Like, mask out the one already found and repeat the `numberOfTrailingZeros` call, until it returns 64.

Reply:
Tried that in 710d0c081a4bb98d7e4cb29edfbacc3dc7a4dbbe

Reply:
I made prototype code like this:

```java
long word = 0x0102030005000708L;
long zeroBytesWord = checkZeroByte(word);
int numTrailingZeros = Long.numberOfTrailingZeros(zeroBytesWord);
int foundIndex = numTrailingZeros / 8;
while (foundIndex < 8) {
    zeroBytesWord >>= numTrailingZeros + 1;
    numTrailingZeros = Long.numberOfTrailingZeros(zeroBytesWord);
    System.out.format("%d\n", foundIndex);
    foundIndex += 1 + numTrailingZeros / 8;
}
```

Do you think it's possible to adapt it to our loop?

Reply:
The current loop is simpler while it has the same effect:
```java
                    for (int j = 0; j < 8; j++) {
                        // Check if the pattern matches only for matched first bytes.
                        if ((check & SwarUtils.ZERO_CHECK_BYTE) != 0 && (us.longAt(i + j) & patternMask) == patternWord) {
                            return true;
                        }
                        check >>>= 8;
                    }
```

In any case, I'll try your variant.

Reply:
My thinking was to reduce the number of iterations to the necessary minimum. But I don't know if it actually pays off.

Reply:
Applied the suggestion here: 105ea9bdc0a775b71bae3b0fe51f5ed2804f5784

Reply:
Good idea. I'll do the measurements a bit later and post them here.

Reply:
Looks like the complexity is worth it in the worst-case scenario, so we should keep it.
```sql
create table xyz as (
  select '00000008000000080000000800000008000000080000000800000008000000080000000800000008000000080000000800000008' s,
         timestamp_sequence(0, 10000) ts
  from long_sequence(10000000)
) timestamp(ts) partition by DAY;

-- before: 330ms
-- after: 270ms
select count(*) from xyz where s like '%88888888%';
```

Reply:
I've added first byte checks to tail in c71a7dc0834cd5688a78f72221fc5e0275e8238b

I also tried more complicated approaches, but this one appears to be the fastest. PTAL